### PR TITLE
Starfield: some BFCB work

### PR DESCRIPTION
**Pull request by jbostrus**

### CODE REVIEW

File: Core/wbDefinitionsCommon.pas [hunk 1]
```diff
@@ -955,6 +961,21 @@ procedure wbATANsAfterSet(const aElement: IwbElement; const aOldValue, aNewValue
...
+procedure wbBODCsAfterSet(const aElement: IwbElement; const aOldValue, aNewValue: Variant);
+begin
+  wbCounterAfterSet('BODM - Count', aElement);
+end;
+
+procedure wbEXASsAfterSet(const aElement: IwbElement; const aOldValue, aNewValue: Variant);
+begin
+  wbCounterAfterSet('EXAC - Count', aElement);
+end;
+
```
Comment (ElminsterAU) [inline Core/wbDefinitionsCommon.pas:978]: I'll have to think about a better way to do this than defining countless callbacks... should be possible to specify simply the signature of the counter subrecord on the RArray definition...

File: Core/wbDefinitionsSF1.pas [hunk 2]
```diff
@@ -9115,32 +9115,45 @@ procedure DefineSF1b;
...
                 wbFloat('Z')
               ]),
-              wbStruct('Unknown', [
-                wbFloat,
-                wbFloat,
-                wbFloat
+              wbStruct('Rotation', [
+                wbFloat('X'),
+                wbFloat('Y'),
+                wbFloat('Z')
```
Comment (ElminsterAU) [inline Core/wbDefinitionsSF1.pas]: What units are the rotations specified in? if it isn't degrees, could use the same approach as is used in REFR/DATA

File: Core/wbDefinitionsSF1.pas [hunk 3]
```diff
@@ -9209,15 +9222,21 @@ procedure DefineSF1b;
...
+            )
           ], []),
-          wbUnknown(EXAC),
-          wbRArray('Unknown', wbString(EXAS)),
-          wbUnknown(EXBS)
+        wbRstruct('Unknown', [
+          wbInteger(EXAC, 'Count', itU32), // count for EXAS array
+          wbRArray('Unknown', wbString(EXAS), cpNormal, False, nil, wbEXASsAfterSet)
+        ], []),
+          wbString(EXBS).SetRequired(True)
```
Comment (ElminsterAU) [inline Core/wbDefinitionsSF1.pas]: something looks wrong with the indendation here?

File: Core/wbDefinitionsSF1.pas [hunk 4]
```diff
@@ -9311,7 +9330,11 @@ procedure DefineSF1b;
         ], []),
         //BGSSpaceshipWeaponBindings_Component
         wbRStruct('Component Data', [
-          wbUnknown(SHWB)
+          wbStruct(SHWB, 'Ship Weapon Binding', [
+            wbInteger('Weapon Slot 1', itS32),
+            wbInteger('Weapon Slot 2', itS32),
+            wbInteger('Weapon Slot 3', itS32)
+          ])
```
Comment (ElminsterAU) [inline Core/wbDefinitionsSF1.pas:9331]: what exactly do these integers represent?

**Review by ElminsterAU — CHANGES_REQUESTED** [review]

File: Core/wbDefinitionsSF1.pas (hunk 3 above)
Comment (jbostrus) [reply Core/wbDefinitionsSF1.pas]: yes

File: Core/wbDefinitionsSF1.pas (hunk 4 above)
Comment (jbostrus) [reply Core/wbDefinitionsSF1.pas:9331]: integers relate to the ID of the item from the components list

File: Core/wbDefinitionsSF1.pas (hunk 2 above)
Comment (jbostrus) [reply Core/wbDefinitionsSF1.pas]: radians

**Pull request merged**